modules/consensus.go

```diff
@@ -260,3 +267,32 @@ func (cc ConsensusChange) Append(cc2 ConsensusChange) ConsensusChange {
 		DelayedSiacoinOutputDiffs: append(cc.DelayedSiacoinOutputDiffs, cc2.DelayedSiacoinOutputDiffs...),
 	}
 }
+
+// String prints the ConsensusChangeID in hex.
+func (ccid ConsensusChangeID) String() string {
+	return hex.EncodeToString(ccid[:])
+}
+
+// MarshalJSON marshales a ConsensusChangeID as a hex string.
+func (ccid ConsensusChangeID) MarshalJSON() ([]byte, error) {
+	return json.Marshal(ccid.String())
+}
+
+// UnmarshalJSON decodes the json hex string of the ConsensusChangeID.
+func (ccid *ConsensusChangeID) UnmarshalJSON(b []byte) error {
+	// *2 because there are 2 hex characters per byte.
+	// +2 because the encoded JSON string has a `"` added at the beginning and end.
+	if len(b) != crypto.HashSize*2+2 {
+		return crypto.ErrHashWrongLen
+	}
+
+	// b[1 : len(b)-1] cuts off the leading and trailing `"` in the JSON string.
+	hBytes, err := hex.DecodeString(string(b[1 : len(b)-1]))
+	if err != nil {
+		return errors.New("could not unmarshal ConsensusChangeID: " + err.Error())
+	}
+	copy(ccid[:], hBytes)
+	return nil
```

Comment:
Is there a way in Go to avoid this by calling the marshalling functions of the `crypto.Hash` ? I did some searching but it seems that named types in Go are strictly separated and don't like casting.